server/multitenancy/test/tenant_resolver.test.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { resolve } from '../tenant_resolver';
+
+describe("Resolve tenants when multitenancy is enabled and both 'Global' and 'Private' tenants are disabled", () => {
+  function resolveWithConfig(config: any) {
+    return resolve(
+      config.username,
+      config.requestedTenant,
+      config.preferredTenants,
+      config.availableTenants,
+      config.globalTenantEnabled,
+      config.privateTenantEnabled
+    );
+  }
+
+  it('Resolve tenants list for admin user', () => {
+    const adminConfig = {
+      username: 'admin',
+      requestedTenant: 'admin_tenant',
+      preferredTenants: undefined,
+      availableTenants: { global_tenant: true, admin_tenant: true, test_tenant: true, admin: true },
+      globalTenantEnabled: false,
+      privateTenantEnabled: false,
+    };
+
+    const adminResult = resolveWithConfig(adminConfig);
+    expect(adminResult).toEqual('admin_tenant');
+  });
+
+  it('Resolve tenants list for non-admin user', () => {
```

Comment:
Do these tests have anything to do with whether the user is admin or non-admin?

What do you think about:

1. Resolve tenant with custom tenants, Global and Private disabled
2. Resolve tenant without custom tenants, Global and Private disabled

Reply:
Yes, that makes sense, I was using this because as the admin user I think you will have the `admin_tenant` as one of the reserved custom tenants, but you can always change that.

Reply:
Done!